### ISSUE TITLE
🎁 Include closed captions / transcripts in manifest

### ISSUE THREAD

**Issue by kirkkwang** [notch8/utk-hyku#442]
# Story

UTK want the transcripts of A/V works on their IIIF manifests.

related to: 
- [x] #483

# Acceptance Criteria

- [ ] Audio and video IIIF manifests should have transcripts

# Screenshots / Video

<img width="860" alt="image" src="https://github.com/scientist-softserv/utk-hyku/assets/19597776/724cdf1f-30cf-4892-abee-6677fcbe0dba">

# Testing Instructions and Sample Files

- [ ] import this csv ~~[rfta_test_with_attachments_video.csv](https://github.com/scientist-softserv/utk-hyku/files/11837804/rfta_test_with_attachments_video.csv)~~ UPDATED FILE: [rfta_test_with_attachments_video(1).csv](https://github.com/scientist-softserv/utk-hyku/files/11885364/rfta_test_with_attachments_video.1.csv)

  - NOTE: the remote files were replaced with something smaller for testing purposes
- [ ] Navigate to the work's /manifest.json route
- [ ] Observe that the `annotations` field with the transcripts (in this example there should be an English and Spanish)

# Notes

At this point in time the video may not be viewable unless the representative and thumbnail are set manually in the work's edit page.
- https://github.com/scientist-softserv/utk-hyku/issues/438

**Comment by jillpe** [comment]
SoftServ QA: 

csv imported: 

[rfta_test_with_attachments_video.csv](https://github.com/scientist-softserv/utk-hyku/files/11871685/rfta_test_with_attachments_video.csv)



[Import URL](https://dc.utk-hyku-staging.notch8.cloud/importers/134?locale=en)


[/manifest.json](https://dc.utk-hyku-staging.notch8.cloud/concern/videos/bbc34b4b-e0ff-48f9-8a54-db3d68ac3ede/manifest.json)

Annotations: only english is present in the annotations field

![Image](https://github.com/scientist-softserv/utk-hyku/assets/84697174/6503c544-f8aa-4867-ae07-790c5ab24068)

**Comment by jillpe** [comment]
related to
- #446

**Comment by ShanaLMoore** [comment]
altered csv: [442.csv](https://github.com/scientist-softserv/utk-hyku/files/11899839/442.csv)

the original CSV's Video was referencing identifiers to collections that didn't exist, which was the reason the create relationships jobs kept failing. 

For some reason, the spanish transcript files [did not import](https://dc.utk-hyku-staging.notch8.cloud/importers/135/entries/15135?locale=en) which is the reason Jill didn't see it in the manifest.

**Comment by ShanaLMoore** [comment]
@jillpe 

I deleted the original importer and the video file it created, and reimported the following altered file: [442.csv](https://github.com/scientist-softserv/utk-hyku/files/11885822/442.csv)

The file that you used was trying to reference a collection that didn't exist, so the create relationship jobs never succeeded. I'm thinking about a way we can raise a better error, but that shouldn't stop you from redoing QA. Maybe try this again using a [different tenant](https://qa.utk-hyku-staging.notch8.cloud/) if you want to retest from scratch? 

Updated importer: https://dc.utk-hyku-staging.notch8.cloud/importers/135?locale=en

I made sure all entries were successful: 

Work entries: 
![image](https://github.com/scientist-softserv/utk-hyku/assets/10081604/3f7de475-3c76-4352-823a-782efaad4505)

FileSet entries: 
![image](https://github.com/scientist-softserv/utk-hyku/assets/10081604/949109b2-aa25-498f-8d3d-091e9214bd76)


~~I'm not sure if I should expect the [video to play in the UV ](https://dc.utk-hyku-staging.notch8.cloud/concern/videos/7f7274f3-ff34-422c-b4e6-685748423266?locale=en)at this point, but that doesn't seem to be an acceptance criteria of the ticket anyways.~~

edit: It plays just fine in chrome. My above comment wasn't playing in firefox. 🤔 



Afterwards I was able to visit the video record's /[manifest.json](https://dc.utk-hyku-staging.notch8.cloud/concern/videos/7f7274f3-ff34-422c-b4e6-685748423266/manifest.json) and see the spanish transcript in there: 

![image](https://github.com/scientist-softserv/utk-hyku/assets/10081604/be9090ca-cc78-4cdb-807e-489b548c87ff)

auto generated thumbnail: 

<img width="1098" alt="image" src="https://github.com/scientist-softserv/utk-hyku/assets/10081604/d0741b9c-3761-4226-826b-58d1dc67068a">

**Comment by jillpe** [comment]
SoftServ QA: Pass

<details><summary>First Test</summary>
csv:

[rfta_test_with_attachments_video.csv](https://github.com/scientist-softserv/utk-hyku/files/11909342/rfta_test_with_attachments_video.csv)



[Importer URL](https://qa.utk-hyku-staging.notch8.cloud/importers/36?locale=en)

[Manifest URL](https://qa.utk-hyku-staging.notch8.cloud/concern/videos/1f0e2ba1-acf2-4508-8b3c-9aa331e984df/manifest.json)
- Manifest shows 1 language (es) in the `annotations` field which is correct. 
</details>


<details><summary>Second Test</summary>
csv: 

[rfta_test_with_attachments_video.1.csv](https://github.com/scientist-softserv/utk-hyku/files/11909344/rfta_test_with_attachments_video.1.csv)

[Importer URL](https://qa.utk-hyku-staging.notch8.cloud/importers/36?locale=en)

[Manifest URL](https://qa.utk-hyku-staging.notch8.cloud/concern/videos/b641088a-2e95-4735-86d3-b9f19f0867bc/manifest.json)
- shows 2 languages (en and es), which is correct

![Image](https://github.com/scientist-softserv/utk-hyku/assets/84697174/c836eec7-39b0-44a7-95d2-ff06726bd8ee)



</details>

**Comment by markpbaggett** [comment]
Looks good here:

https://dc.utk-hyku-staging.notch8.cloud/concern/videos/febc6b41-d998-47ab-9732-ac09fd8168f2/manifest

and here:

https://dc.utk-hyku-staging.notch8.cloud/concern/audios/b648cfc2-b87f-43e1-ac85-a87a6d32978d/manifest

**Comment by jillpe** [comment]
Verification on production blocked until #358 resolved

**Comment by jillpe** [comment]
UTK will verify this when they start bringing in audio imports

**Comment by markpbaggett** [comment]
@jillpe and @kirkkwang something odd is happening here in production.  Take a look at [Bopp, William F.: WWII Oral History - Part 2](https://dc.utk-hyku-production.notch8.cloud/concern/audios/1cdbe3d9-53eb-4949-adb6-ca50e97fb0de).  It clearly has captions:



![Image](https://github.com/scientist-softserv/utk-hyku/assets/2692416/b32814b8-0953-450d-a897-cf845bffdcca)

Also, those captions look like they are modeled correctly in the manifest:

![Image](https://github.com/scientist-softserv/utk-hyku/assets/2692416/476f9e0b-4187-4463-8c1f-dfe1fa8b1644)

But when you access them, you get a Rails error:

![Image](https://github.com/scientist-softserv/utk-hyku/assets/2692416/4632391e-83c9-44e2-bd87-ec61b085aaad)

Any idea what's happening here?

**Comment by kirkkwang** [comment]
@markpbaggett that's very strange why it doesn't show up, here's the vtt download that it should pull from
https://dc.utk-hyku-production.notch8.cloud/downloads/74b91038-3fc3-436d-9d44-87a308dea6c7

I don't think that id that you clicked on ever worked in hyrax applications, but the link above is where the viewer should pull the caption from which downloads, but I'm not sure off the top of my head why it's not going into the viewer

I do see this in the console:
<img width="1170" alt="image" src="https://github.com/scientist-softserv/utk-hyku/assets/19597776/fbf35a37-7e7d-4541-a13b-2fc3b62451ec">

The strange thing is that could have sworn this was working before.

<img width="262" alt="image" src="https://github.com/scientist-softserv/utk-hyku/assets/19597776/5fc16dbd-7372-47ab-9439-c9de60cb828f">

**Comment by josh-morgan117** [comment]
I don't see the captions appearing in the viewer when I click on "Captions in English". 

Also, with this example we are still seeing the language as a URI.

**Comment by mlhale7** [comment]
Captions are showing after @kirkkwang updated an individual record - https://digitalcollections.lib.utk.edu/concern/videos/225b11f6-e889-4406-a4d8-e258dc4d4661 A full reindex is needed to see this change across all works.

<img width="1189" alt="Image" src="https://github.com/user-attachments/assets/2d0fdf6a-4737-4682-b6cd-11bf42d1f6da" />

**Comment by josh-morgan117** [comment]
@kirkkwang Looks like audio works aren't showing captions. https://digitalcollections.lib.utk.edu/concern/compound_objects/1bb2cd90-9f41-4e26-8237-6874631b1594?locale=en

**Comment by kirkkwang** [comment]
@josh-morgan117 i actually don't see the transcript files as a fileset, could they have been missed?

<img width="1103" alt="Image" src="https://github.com/user-attachments/assets/48cdabb5-0233-4696-ad16-2541d3c816fc" />

**Comment by josh-morgan117** [comment]
I always find the problem records by chance. I'll check
